plugins/websocket_server/WebsocketServer.cc

```diff
@@ -40,22 +40,152 @@ using namespace ignition::launch;
+  int n;
+
+  p = buf + LWS_PRE;
+  start = p;
+  end = p + sizeof(buf) - LWS_PRE;
+
+  // Status code
+  if(lws_add_http_header_status(_wsi,
+      status_code,
+      reinterpret_cast<unsigned char **>(&p),
+      end))
+    return 1;
+
+  // Content-Type
+  if (lws_add_http_header_by_token(_wsi,
+      WSI_TOKEN_HTTP_CONTENT_TYPE,
+      reinterpret_cast<const unsigned char *>(content_type),
+      strlen(content_type),
+      &p,
+      end))
+    return 1;
+
+  // Content-Length
+  if (lws_add_http_header_content_length(_wsi,
+      content_length-1,
+      &p,
+      end))
+    return 1;
+
+  // Finalize header
+  if (lws_finalize_http_header(_wsi, &p, end))
+    return 1;
+
+  // Write headers
+  n = lws_write(_wsi, start, p - start, LWS_WRITE_HTTP_HEADERS);
+  if (n < 0)
+    return 1;
+
+  return 0;
+}
+
+int httpCallback(struct lws *_wsi,
+                 enum lws_callback_reasons _reason,
+                 void * user,
+                 void * _in,
+                 size_t len) {
+  WebsocketServer *self = get_server(_wsi);
+
+  switch (_reason) {
+    case LWS_CALLBACK_HTTP:
+    {
+      char *URI = (char *) _in;
+      igndbg << "Requested URI: " << URI << "\n";
+
+      // Router
+      // Server metrics
+      if (strcmp(URI, "/metrics") == 0) {
+        igndbg << "Handling /metrics\n";
+
+        // TODO Support a proper way to output metrics
+
+        // Format contains the format of the string returned by this route.
+        // The following metrics are currently supported:
+        // * connections - Number of live connections.
+        const char *format = "{ \"connections\": %s }";
+
+        // Get number of connections
+        std::string conns = std::to_string(self->connections.size());
+
+        // Prepare the output
+        size_t buflen = strlen(format) + (conns.size() - 1);
+        unsigned char buf[buflen + LWS_PRE];
+        int n;
+        n = snprintf(reinterpret_cast<char *>(buf), buflen, format,
+            conns.c_str());
+        // Check that no characters were discarded
+        if (n - int(buflen) > 0) {
+          ignwarn << "Discarded "
+            << n - int(buflen)
+            << "characters when preparing metrics.\n";
+        }
+
+        // Write response headers
+        if (write_http_headers(_wsi, 200, "application/json", buflen))
+          return 1;
+
+        // Write response body
+        lws_write_http(_wsi,
+                       reinterpret_cast<unsigned char *>(buf),
+                       strlen(reinterpret_cast<const char *>(buf)));
+        break;
+      }
+      // Return a 404 if no route was matched
+      else {
+        igndbg << "Resource not found.\n";
+        lws_return_http_status(_wsi, HTTP_STATUS_NOT_FOUND, "Not Found");
+      }
+      break;
+    }
+
+    default:
+      // Do nothing on default.
+      break;
+  }
+
+  return -1;
+}
+
+int rootCallback(struct lws *_wsi,
+                 enum lws_callback_reasons _reason,
+                 void * user,
```

Comment:
```suggestion
                 void *_user,
```

Reply:
Addressed in https://github.com/ignitionrobotics/ign-launch/pull/49/commits/1fee42e8a1c08937f8610a1581d5926efdb396f1.